sdk/core/azure-core/inc/azure/core/cryptography/hash.hpp

```diff
@@ -135,7 +136,7 @@ namespace Azure { namespace Core { namespace Cryptography {
     ~Md5Hash() override;
 
   private:
-    void* m_md5Context;
+    std::unique_ptr<Hash> m_implementation;
```

Comment:
It works, but is it a common design pattern?

Reply:
Well, @CaseyCarter  might comment about it.  I don't really know if that's a pattern.

I just wanted to avoid using void* and all the dynamic_casting by introducing a second Abstraction layer for the OS